src/kaboom.ts

```diff
@@ -2159,16 +2163,34 @@ function state(initState: string, stateList?: string[]): StateComp {
 		events[state][event].forEach((action) => action(...args));
 	}
 
+	function checkTransition(from: string, to: string) {
+		if (!transitions?.[from]) {
+			return
+		}
+		const available = typeof transitions[from] === "string"
+			? [transitions[from]]
+			: transitions[from] as string[];
+		if (!available.includes(to)) {
+			throw new Error(`Cannot transition state from "${from}" to "${state}". Available transitions: ${available.map((s) => `"${s}"`).join(", ")}`);
+		}
```

Comment:
I wonder if it's more friendly if we return a boolean indicating success or failure.